Implement rest api to get permission metadata [#33394]

hassandotcms opened the pull request:
### Proposed Changes

Implemented REST API to fetch permissions metadata (supported permission types and levels)

This PR fixes: #33394

This PR fixes: #33394

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- add new response view class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
- implement helper methods for user permission rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
 - get api implementation to return user specific permissions by id/email, by hosts/folder.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- integration test added to verify the functionality of get user permissions rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

add postman test.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

fix integration tests, response type error.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

support api classes to be available for injection via CDI.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

enable CDI for userPermissionHelper, to enable injection.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 fix integ tests

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

  - Return specific response type instead of generic Response
  - Add better documentation and error messages
  - Split user loading into separate helper method

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 refactoring and cleanup.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

move helper method to helper class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 add view class for permission metadata.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

implement rest api for permission metadata.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

add javadoc, refactoring in helper methods to make them reusable and update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 fix constructor init issue.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 add postman test for permission metadata api

jdotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java:
```diff
@@ -1334,4 +1374,82 @@ public final void delete(@Context final HttpServletRequest httpServletRequest,
+		final List<Map<String, Object>> permissions = userPermissionHelper
+			.buildUserPermissionResponse(userRole, requestingUser);
+
+		final Map<String, Object> responseData = Map.of(
```
Again, I would prefer a bean rather than a Map

hassandotcms replied:
resolved

jdotcms reviewed (approved):
Approve with some comments

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(users-portlet) #33394

 - Create PermissionMetadata, UserPermissions, and UserPermissionAsset beans for type-safe API responses
 - Update permission endpoints to return ResponseEntityView wrappers instead of generic Response objects
 - Replace Map-based data structures with beans

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/system/permission/PermissionResource.java:
```diff
@@ -268,6 +276,46 @@ public Response getByContentletGroupByType(final @Context HttpServletRequest req
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getPermissionMetadata(@Parameter(hidden = true) @Context HttpServletRequest request,
```
done

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(users-portlet): rest api for permission metadata #33394

update auth for permission api.

hassandotcms committed:
Merge branch 'main' into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
refactor(permissions): align metadata API with specification #33394

- Change endpoint path from /metadata to / (root collection path)
  - Update permission scope terminology from STRUCTURE to CONTENT_TYPE
  - Add 403 Forbidden response documentation to OpenAPI annotations
  - Update Postman collection to use new endpoint path

hassandotcms committed:
api doc update

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
reorganize files and sync with main. #33394

hassandotcms committed:
modernize(permissions): refactor utils. #33394

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
remove unrelated user package changes #33394

hassandotcms committed:
Merge branch 'main' into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(permission): rest api to get permission metadata [#33394]

- use immutables for views. AbstractPermissionMetadataView

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(permissions): api to get permissions metadata [#33394]

- use immutables
- move relevant methods to PermissionAPI
